src/str/basics.rs

```diff
@@ -288,3 +288,63 @@ where
         pattern: pattern.into(),
     }
 }
+
+/// Predicate that checks for all patterns.
+///
+/// This is created by `predicates::str:contains_all`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ContainsAllPredicate {
+    patterns: Vec<String>,
+}
+
+impl Predicate<str> for ContainsAllPredicate {
+    fn eval(&self, variable: &str) -> bool {
+        for pattern in &self.patterns {
+            if !variable.contains(pattern) {
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+impl reflection::PredicateReflection for ContainsAllPredicate {}
+
+impl fmt::Display for ContainsAllPredicate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let palette = crate::Palette::new(f.alternate());
+        write!(
+            f,
+            "{}.{}({})",
+            palette.var("var"),
+            palette.description("contains_all"),
+            palette.expected(format!("{:?}", &self.patterns)),
+        )
+    }
+}
+
+/// Creates a new `Predicate` that ensures a str contains `pattern`
+///
+/// # Examples
+///
+/// ```
+/// use predicates::prelude::*;
+///
+/// let predicate_fn = predicate::str::contains_all(vec!["One", "Two", "Three"]);
+/// assert_eq!(true, predicate_fn.eval("One Two Three"));
+/// assert_eq!(false, predicate_fn.eval("One Two Four"));
+/// assert_eq!(false, predicate_fn.eval("Four Five Six"));
+/// ```
```

Comment:
Using multiple `contains` for a small number of strings to check is fine. This, however, becomes cumbersome when dealing with 10+ different strings to check. Here it would be neat to be able to check a list of strings, either coming from a file or hardcoded ones in the test case.

Reply:
In a way, this is a special casing of other predicates (anding a bunch of contains) and I'm trying to understand when we are justified in baking that in vs having people build it up from other parts.

Speaking of, I wonder if we should implement `Predicate` for tuples for easier and-ing.

Reply:
> Speaking of, I wonder if we should implement `Predicate` for tuples for easier and-ing.

Yes, that could work. However, this again breaks when the user reaches N, with N being the maximum number of items in a tuple we implemented `Predicate` for.

Reply:
Tuples can be composed of tuples.

And for a case like this, the types will all be the same, so we could also support arrays